UI/MemStickScreen.cpp

```diff
@@ -25,23 +25,83 @@
 #include "Common/UI/ViewGroup.h"
 
 #include "Common/StringUtils.h"
-#include "Common/File/FileUtil.h"
-#include "Common/File/Path.h"
 #include "Common/System/System.h"
 #include "Common/System/NativeApp.h"
-#include "Common/Data/Text/I18n.h"
 #include "Common/System/Display.h"
+#include "Common/Data/Text/I18n.h"
+
+#include "Common/File/AndroidStorage.h"
+#include "Common/File/FileUtil.h"
+#include "Common/File/Path.h"
+#include "Common/File/DiskFree.h"
 
 #include "Core/Util/GameManager.h"
 #include "Core/System.h"
+#include "Core/Config.h"
 
 #include "UI/MemStickScreen.h"
 #include "UI/MainScreen.h"
 #include "UI/MiscScreens.h"
 
-#include "Core/Config.h"
+static bool FolderSeemsToBeUsed(Path newMemstickFolder) {
+	// Inspect the potential new folder.
+	if (File::Exists(newMemstickFolder / "PSP") || File::Exists(newMemstickFolder / "SYSTEM")) {
+		// Does seem likely. We could add more critera like checking for actual savegames or something.
+		return true;
+	} else {
+		return false;
+	}
+}
+
+static bool SwitchMemstickFolderTo(Path newMemstickFolder) {
+	Path testWriteFile = newMemstickFolder / ".write_verify_file";
+
+	// Doesn't already exist, create.
+	// Should this ever happen?
+	if (newMemstickFolder.Type() == PathType::NATIVE) {
+		if (!File::Exists(newMemstickFolder)) {
+			File::CreateFullPath(newMemstickFolder);
+		}
+		if (!File::WriteDataToFile(true, "1", 1, testWriteFile)) {
+			// settingInfo_->Show(sy->T("ChangingMemstickPathInvalid", "That path couldn't be used to save Memory Stick files."), nullptr);
+			// TODO: Display an error!
+			return UI::EVENT_DONE;
+		}
+		File::Delete(testWriteFile);
+	} else {
+		// TODO: Do the same but with scoped storage? Not really necessary, right? If it came from a browse
+		// for folder, we can assume it exists and is writable, barring wacky race conditions like the user
+		// being connected by USB and deleting it.
+	}
+
+	Path memStickDirFile = g_Config.internalDataDirectory / "memstick_dir.txt";
+	std::string str = newMemstickFolder.ToString();
+	if (!File::WriteDataToFile(true, str.c_str(), (unsigned int)str.size(), memStickDirFile)) {
+		ERROR_LOG(SYSTEM, "Failed to write memstick path '%s' to '%s'", newMemstickFolder.c_str(), memStickDirFile.c_str());
+		// Not sure what to do if this file.
+	}
+
+	// Save so the settings, at least, are transferred.
+	g_Config.memStickDirectory = newMemstickFolder;
+	g_Config.SetSearchPath(GetSysDirectory(DIRECTORY_SYSTEM));
+	g_Config.UpdateIniLocation();
+
+	return true;
+}
+
+static std::string FormatSpaceString(int64_t space) {
+	if (space >= 0) {
+		// TODO: Smarter display (MB, GB as appropriate). Don't we have one of these somewhere?
```

Comment:
Probably should be in Common:
https://github.com/hrydgard/ppsspp/blob/0c40e918c92b897f745abee0d09cf033a1572337/Windows/W32Util/Misc.cpp#L56

But also:
https://github.com/hrydgard/ppsspp/blob/87dcf60ed037d5f2168d5f0033dda002b098e177/Core/Dialog/SavedataParam.cpp#L1050

-[Unknown]

Reply:
Right.. I'll just move NiceSizeFormat out to Common for now, could merge them later but they're slightly different (rounding, decimal...)